social_core/tests/backends/test_orcid.py

```diff
@@ -0,0 +1,25 @@
+import json
+
+from .oauth import OAuth2Test
+
+from social_core.backends.orcid import ORCIDOAuth2
+
+class ORCIDOAuth2Test(OAuth2Test):
+    backend_path = 'social_core.backends.orcid.ORCIDOAuth2'
+    user_data_url = 'https://login.orbi.kr/oauth/user/get'
```

Comment:
This should probably match USER_ID_URL...

Reply:
@nijel Ahhhh random. Will take a look.

Reply:
> Sorry for not being clear - Orbi is the last one to pass, then the test hangs, so my assumption is that its ORCID that hangs...

Yeh that's probably a fair shout, it's the one alphabetically before ORCID.

Reply:
> You don't need to use tox and test all Python versions.
> 
> Either run tox py38 just for the version you use.
> 
> Or install needed deps directly via `pip install -r social_core/tests/requirements.txt` and then run `pytest` with your test file.



> You don't need to use tox and test all Python versions.
> 
> Either run tox py38 just for the version you use.
> 
> Or install needed deps directly via `pip install -r social_core/tests/requirements.txt` and then run `pytest` with your test file.

For both of these actions I'm getting a python wheels error: `ERROR: Could not build wheels for xmlsec which use PEP 517 and cannot be installed directly`

Reply:
You will have to install xmlsec headers for that, on Ubuntu/Debian it's called `libxmlsec1-dev`.

Reply:
This is where you mock the responses of the service, it should not talk to the real service for the tests. For example I've added OAuth2 tests for Dropbox in https://github.com/python-social-auth/social-core/pull/557/files

Reply:
@nijel We have looked at the testing functionality from around the app - unfortunately ORCID does not structure things in the same way as other standard oauth backends, to get the ORCID id to query their API we need to make an oauth/userinfo call first. The python-social-auth application is returning a generated username -> so we can not test our particular method unfortunately. Please advise.

Reply:
@nijel So we find the following:

```
FAILED social_core/tests/backends/test_orcid.py::ORCIDOAuth2Test::test_login - AssertionError: 'e07051f297fc4282a36bf05cb262b4d8' != '0000-0002-2601-8132'
FAILED social_core/tests/backends/test_orcid.py::ORCIDOAuth2Test::test_partial_pipeline - AssertionError: '6bcf2f9f1e4247e9b6e891d1f08073b0' != '0000-0002-2601-8132'
```

Reply:
I think all OAuth backends work more or less same - you do OAauth, get some kind of ID and OAuth token and do additional request to get more information about the user. This can be easily covered by the tests....

Reply:
@nijel We're struggling to understand the testing then ... we need more advise on how to test the changes...